compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -637,21 +636,22 @@ private void checkErrors(TypeDefinition unresolvedType, BLangType currentTypeNod
                     // Cyclic dependency detected. We need to add the `unresolvedTypeNodeName` or the
                     // `memberTypeNodeName` to the end of the list to complete the cyclic dependency when
                     // printing the error.
-                    visitedNodes.add(currentTypeNodeName);
+                    visitedNodes.push(currentTypeNodeName);
                     dlog.error((DiagnosticPos) unresolvedType.getPosition(), DiagnosticCode.CYCLIC_TYPE_REFERENCE,
                             visitedNodes);
                     // We need to remove the last occurrence since we use this list in a recursive call.
                     // Otherwise, unwanted types will get printed in the cyclic dependency error.
```

Comment:
If possible, shall we update these comments as well? Or at least open an issue for it?

Reply:
created an issue: #23825